### ISSUE TITLE
Perturbing text by removing list elements containing sentences instead of words in raw text

### ISSUE THREAD

**Issue by alexandrahotti** [marcotcr/lime#181]
First of all thank you for an awesome tool! Now onto my issue.
Instead of inputing a raw text string and removing words and seeing what words have the biggest impact, I tried modifying LIME so that it takes as input a list containing sevral sentences as each element in that list. For instance: 

> document_list=["Darkness at the break of noon. Shadows even the silver spoon. The handmade blade, the child’s balloon. Eclipses both the sun and moon.","To understand you know too soon. There is no sense in trying. Pointed threats, they bluff with scorn.", "Suicide remarks are torn. From the fool’s gold mouthpiece the hollow horn. Plays wasted words, proves to warn. That he not busy being born is busy dying."]

The code then removes one element from that list at a time. These lists will contain approximately 10-20 elements. My issue is that lime is telling me that the most important elements in my lists have really small impact on the classification with values between 2.0*10^(-6) and 0.0046. Thus I am wondering wheter or not I've changed the code incorrectly in someway so taht lime is not working in a correct manner.

This is how i changed the code (lime_text.py):
To the normal input i added: 
1. model: which is a trained keras neural network
2. return_w2v_model(): which returns a dict containg pretrained w2v-vectors.

> exp = explainer.explain_instance(return_w2v_model(),model,dp_lime, model_predict_probability, num_features=6, top_labels=1)

I made these changes as model_predict probability needs booth the w2v- vectors and the neural network model. Here is how model predict probility looks like:

> def model_predict_probability(rawtext,model,w2v_model):
>     newVecs = np.empty((0,3), int) #3 is the numbers of categories in the classification
>     for text in rawtext:
>         doc = preprocess(text)
>         document_vec = document_vector(w2v_model, doc)
>         newVecs = np.append(newVecs, model.predict(np.array([document_vec])), axis = 0)#, batch_size=None, verbose=0, steps=None))
>     return newVecs
> 

> def document_vector(word2vec_model, doc):
> """" Here i create a word2vec centroid for each text document """
>     doc = [word for word in doc if word in word2vec_model.vocab]
>     return np.mean(word2vec_model[doc], axis=0)
> 
> 

Then this is how i changed the class IndexedString i lime_text.py:

> class IndexedString(object):
>     """String with various indexes."""
> 
>     def __init__(self, raw_string_list, split_expression=r'\W+', bow=True):
> 
>         self.raw = " ".join(raw_string_list)
>         self.as_list=raw_string_list
>         self.as_np = np.array(self.as_list)
> 
>         sentences_span=[]
>         start=0
>         end=0
>         for text_part in self.as_list:
>             end=len(text_part)+start
>             sentences_span.append((start,end))
>             start=end+1
>         non_word = re.compile(r'(%s)|$' % split_expression).match
>         self.string_start = np.array([begin for (begin, end) in sentences_span])

Thank you in advance!

**Comment by marcotcr** [comment]
Hey! No, this approach should be fine. I would have to take a look at the specific model to understand what's going on, but I see you closed the issue. Was there something wrong?

**Comment by alexandrahotti** [comment]
Hi! I closed my issue as i managed to solve the problem. It turns out that I actually had an issue with my preprocessing and not with lime. Thank you for answering!
